File: tickezy_backend/src/models/notification.js
const { DataTypes } = require('sequelize');
const sequelize = require('../../db');

const Notification = sequelize.define('Notification', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  title: DataTypes.STRING,
  message: DataTypes.TEXT,
  type: {
    type: DataTypes.ENUM('TICKET_CONFIRMATION', 'EVENT_REMINDER', 'PAYMENT_SUCCESS', 'EVENT_UPDATE', 'ADMIN_MESSAGE'),
  },
  timestamp: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW,
  },
  isRead: {
    type: DataTypes.BOOLEAN,
    defaultValue: false,
  },
  relatedEventId: DataTypes.UUID,
  relatedTicketId: DataTypes.UUID,
}, {
  timestamps: true,
});

module.exports = Notification;

File: TICKEZY/TICKEZY/Models/Notification.swift
//
//  Notification.swift
//  TICKEZY
//
//  Created by M.A on 10/14/25.
//

import Foundation

enum NotificationType: String, Codable {
    case TICKET_CONFIRMATION, EVENT_REMINDER, PAYMENT_SUCCESS, EVENT_UPDATE, ADMIN_MESSAGE
}

struct Notification: Codable, Identifiable {
    let id: String
    let userId: String
    let title: String
    let message: String
    let type: NotificationType
    let timestamp: Date
    let isRead: Bool
    let relatedEventId: String?
    let relatedTicketId: String?
}

